engine/src/main/java/org/terasology/rendering/dag/nodes/DownSamplerNode.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.nodes;
+
+import org.terasology.assets.ResourceUrn;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.dag.ConditionDependentNode;
+import org.terasology.rendering.dag.stateChanges.BindFBO;
+import org.terasology.rendering.dag.stateChanges.EnableMaterial;
+import org.terasology.rendering.dag.stateChanges.SetInputTextureFromFBO;
+import org.terasology.rendering.dag.stateChanges.SetViewportToSizeOf;
+import org.terasology.rendering.opengl.BaseFBOsManager;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FBOConfig;
+import org.terasology.rendering.opengl.FBOManagerSubscriber;
+
+import static org.terasology.rendering.dag.stateChanges.SetInputTextureFromFBO.FboTexturesTypes.ColorTexture;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+
+/**
+ * Instances of this class take the content of the color attachment of an input FBO
+ * and downsamples it into the color attachment of a smaller output FBO.
+ */
+public class DownSamplerNode extends ConditionDependentNode implements FBOManagerSubscriber {
+
+    private static final String TEXTURE_NAME = "tex";
+    private static final int SLOT_0 = 0;
+    private static final ResourceUrn DOWN_SAMPLER_MATERIAL = new ResourceUrn("engine:prog.downSampler");
+
+    private String performanceMonitorLabel;
+    private BaseFBOsManager outputFBOmanager;
+    private FBOConfig outputFBOconfig;
+    private FBO outputFBO;
+    private Material downSampler;
+
+    /**
+     * Throws a RuntimeException if invoked. Use initialise(...) instead.
+     */
+    @Override
+    public void initialise() {
+        throw new RuntimeException("Please do not use initialise(). For this class use initialise(...) instead.");
```

Comment:
Isn't this the classic case of checked vs. unchecked exceptions? Since the programmer is never meant to run `subClassMethod.initialize()` in this case, by making `Node.initialize()` and `DownSamplerNode.initialize()` throw a child of `Exception` (Maybe `NoSuchMethodException`?), the Java compiler will throw an error if somewhere in the code somebody tries to call the `initialize()` method on a child of `DownSamplerNode`. This will avoid having to watch the program crash and read the stack trace at least once for a future programmer trying to construct a child of `DownSamplerNode` with the wrong constructor.